library/pod-security-policy/seccomp/template.yaml

```diff
@@ -35,52 +35,220 @@ spec:
                 type: string
             allowedProfiles:
               type: array
-              description: "An array of allowed profile values for seccomp annotations on Pods."
+              description: >-
+                An array of allowed profile values for seccomp on Pods/Containers.
+
+                Can use the annotation naming scheme: `runtime/default`, `docker/default`, `unconfined` and/or
+                `localhost/some-profile.json`. The item `localhost/*` will allow any localhost based profile.
+
+                Can also use the securityContext naming scheme: `RuntimeDefault`, `Unconfined`
+                and/or `Localhost`. For securityContext `Localhost`, use the parameter `allowedLocalhostProfiles`
+                to list the allowed profile JSON files.
+
+                The policy code will translate between the two schemes so it is not necessary to use both.
+
+                Putting a `*` in this array allows all Profiles to be used.
+
+                This field is required since with an empty list this policy will block all workloads.
               items:
                 type: string
+            allowedLocalhostFiles:
+              type: array
+              description: >-
+                When using securityContext naming scheme for seccomp and including `Localhost` this array holds
+                the allowed profile JSON files.
+
+                Putting a `*` in this array will allows all JSON files to be used.
+
+                This field is required to allow `Localhost` in securityContext as with an empty list it will block.
   targets:
     - target: admission.k8s.gatekeeper.sh
       rego: |
         package k8spspseccomp
 
         import data.lib.exempt_container.is_exempt
 
-        violation[{"msg": msg, "details": {}}] {
-            metadata := input.review.object.metadata
-            not input_wildcard_allowed(metadata)
-            container := input_containers[_]
-            not is_exempt(container)
-            not input_container_allowed(metadata, container)
-            msg := sprintf("Seccomp profile is not allowed, pod: %v, container: %v, Allowed profiles: %v", [metadata.name, container.name, input.parameters.allowedProfiles])
+        container_annotation_key_prefix = "container.seccomp.security.alpha.kubernetes.io/"
+
+        pod_annotation_key = "seccomp.security.alpha.kubernetes.io/pod"
+
+        naming_translation = {
+        	# securityContext -> annotation
```

Comment:
The src rego has been formatted with `opa fmt` which makes it tabs. I personally prefer spaces for indentation but wanted to keep with the format provided by the tooling. I guess the `make generate` uses spaces by default for indentation in the YAML file.

Do you prefer that I use spaces in the src rego instead of using the format output by `opa fmt` @sozercan?

Reply:
yea just so we align with rest of the templates

Reply:
Pushed tabs -> spaces